app/code/Magento/Sales/Model/Order/ShipmentFactory.php

```diff
@@ -104,7 +104,9 @@ protected function prepareItems(
 
             /** @var \Magento\Sales\Model\Order\Shipment\Item $item */
             $item = $this->converter->itemToShipmentItem($orderItem);
-            if ($orderItem->getIsVirtual() || ($orderItem->getParentItemId() && !$orderItem->isShipSeparately())) {
+            if ($orderItem->getIsVirtual() ||
+                ($orderItem->getProductType() === \Magento\Catalog\Model\Product\Type::TYPE_BUNDLE
```

Comment:
This is incorrect module dependency. Although `bundle` constant is placed in `Catalog` module for historical reasons, `Sales` module must not be aware of bundle product type business logic.

If it's not too hard, please move the `bundle_selection_attributes` part to `Bundle` module as well.

Please cover described case with integration test.